Rubberduck.Refactorings/EncapsulateField/EncapsulateFieldInsertNewCode/EncapsulateFieldCodeBuilder.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Symbols;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rubberduck.Refactorings.EncapsulateField
+{
+    public interface IEncapsulateFieldCodeBuilder
+    {
+        (string Get, string Let, string Set) BuildPropertyBlocks(PropertyAttributeSet propertyAttributeSet);
+        string BuildUserDefinedTypeDeclaration(IObjectStateUDT objectStateUDT, IEnumerable<IEncapsulateFieldCandidate> candidates);
+        string BuildObjectStateFieldDeclaration(IObjectStateUDT objectStateUDT);
+        string BuildFieldDeclaration(Declaration target, string identifier);
+    }
+
+    /// <summary>
+    /// EncapsulateFieldCodeBuilder wraps an ICodeBuilder instance to extend it for the 
+    /// specific needs of an EncapsulateField refactoring action.
+    /// </summary>
+    public class EncapsulateFieldCodeBuilder : IEncapsulateFieldCodeBuilder
+    {
+        private const string FourSpaces = "    ";
```

Comment:
Agreed...IMO indenter settings are the right way to go _especially_ where this (or any refactoring) generates brand new code block(s).  When inserting into existing content - the choice is not quite so clear (at least to me)...`EncapsulateFieldRefactoring` needs to support both use cases.  

So, as I've explored how to address this comment, I've come to the decision-point whether to ctor inject an `IIndenter` interface into the `CodeBuilder` (making an Indenter instance _always_ available), or use method injection to allow callers to dictate the 'indention style'.  

I'm leaning toward method injection since it relieves the `CodeBuilder` of the responsibility to determine _when_ to use the `Indenter` (it simply by checking if an `IIndenter` interface has been provided). e.g., use case: using `ICodeBuilder` to generate a line of code to insert into an existing block - where the caller might prefer to apply an indention style that matches what is currently used within the block.  

The downside of method injection is that it adds, yet another, optional parameter to a handful of the `ICodeBuilder` methods - which already have several.  (e.g., 
```
public bool TryBuildPropertyLetCodeBlock(Declaration prototype, 
               string propertyIdentifier, 
               out string codeBlock,
               Accessibility accessibility = Accessibility.Public,
               string content = null,
               string parameterIdentifier = null,
               IIndenter indenter = null) 
```
Thoughts/Reactions/Opinions?...Time to inject a `struct`/DTO?

Reply:
I'm inclining toward KISS / YAGNI - while it might be cool for the callers to change indentation style, I feel this encourage inconsistent style enforcement. I mean, they should be setting the indenter's behavior before they refactor at all, and if the code they're working doesn't conform already, it should conform (and obey 😉 ) to the current indenter's setting. So I'd always provide to the code builder and tell users to go to the settings if they don't like it.

Reply:
Glad I asked...Admittedly, though I was leaning towards the method injection approach, I had my doubts because it seemed intuitively 'correct' that a `CodeBuilder` object _should_ always have access to the current `Indenter` settings.  The approach you are advocating implies that when editing an existing block (e.g., inserting new members into a `UserDefinedType Declaration`, a brand new declaration block will be generated, indented (with both new and existing UDT members), and replace the existing `UserDefinedType Declaration` in its entirety...This is different than the current implementation, but sounds good to me.

Reply:
integrated use of `IIndenter` in the latest sync